services/subscription.go

```diff
@@ -0,0 +1,285 @@
+package services
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/smartcontractkit/chainlink/logger"
+	"github.com/smartcontractkit/chainlink/store"
+	"github.com/smartcontractkit/chainlink/store/models"
+	"github.com/smartcontractkit/chainlink/store/presenters"
+	"github.com/smartcontractkit/chainlink/utils"
+	"go.uber.org/multierr"
+)
+
+// Descriptive indices of a RunLog's Topic array
+const (
+	EventTopicSignature = iota
+	EventTopicRequestID
+	EventTopicJobID
+)
+
+// RunLogTopic is the signature for the Request(uint256,bytes32,string) event
+// which Chainlink RunLog initiators watch for.
+// See https://github.com/smartcontractkit/chainlink/blob/master/solidity/contracts/Oracle.sol
+var RunLogTopic = common.HexToHash("0x06f4bf36b4e011a5c499cef1113c2d166800ce4013f6c2509cab1a0e92b83fb2")
+
+// Listens to event logs being pushed from the Ethereum Node specific to a job.
+type JobSubscription struct {
+	Job                    models.Job
+	InitiatorSubscriptions []InitiatorSubscription
```

Comment:
Why use an interface if there is only one type used here?

Reply:
TL;DR It's a better practice to define your contracts with an interface. Allows for better testability, by allowing a `MockInitiatorSubscriotion`, something I am not leveraging, and acts as self documenting code by conveying intent. I think it's cleaner if we keep it, but don't feel strongly about it.

Originally, I had different subscription types before pivoting towards the `ReceiveRunLog` and `ReceiveEthLog` callback strategy. Those types were going to comply with this interface so `JobSubscription` wouldn't have to care about the specific type.

We could rename `RpcSubscription` to `InitiatorSubscription` and remove the interface but my intuition makes me reluctant to do it.